homeassistant/components/xiaomi_miio/select.py

```diff
@@ -0,0 +1,208 @@
+
+    name: str = None
+    icon: str = None
+    short_name: str = None
+    options: list = None
+    service: str = None
+
+
+SELECTOR_TYPES = {
+    FEATURE_SET_LED_BRIGHTNESS: SelectorType(
+        name="Led brightness",
+        icon="mdi:brightness-6",
+        short_name=ATTR_LED_BRIGHTNESS,
+        options=["Bright", "Dim", "Off"],
+        service=SERVICE_SET_LED_BRIGHTNESS,
+    ),
+}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Selectors from a config entry."""
+    entities = []
+    if config_entry.data[CONF_FLOW_TYPE] == CONF_DEVICE:
+        host = config_entry.data[CONF_HOST]
+        token = config_entry.data[CONF_TOKEN]
+        model = config_entry.data[CONF_MODEL]
+        device = hass.data[DOMAIN][config_entry.entry_id][KEY_DEVICE]
+        coordinator = hass.data[DOMAIN][config_entry.entry_id][KEY_COORDINATOR]
+        device_features = 0
+        entity_class = None
+
+        _LOGGER.debug("Initializing with host %s (token %s...)", host, token[:5])
+
+        if model in [MODEL_AIRHUMIDIFIER_CA1, MODEL_AIRHUMIDIFIER_CB1]:
+            device_features = FEATURE_FLAGS_AIRHUMIDIFIER_CA_AND_CB
+            entity_class = XiaomiAirHumidifierSelector
+        elif model in [MODEL_AIRHUMIDIFIER_CA4]:
+            device_features = FEATURE_FLAGS_AIRHUMIDIFIER_CA4
+            entity_class = XiaomiAirHumidifierMiotSelector
+        elif model in MODELS_HUMIDIFIER:
+            device_features = FEATURE_FLAGS_AIRHUMIDIFIER
+            entity_class = XiaomiAirHumidifierSelector
+        else:
+            return
+
+        for feature in SELECTOR_TYPES:
+            selector = SELECTOR_TYPES[feature]
+            if feature & device_features and feature in SELECTOR_TYPES:
+                entities.append(
+                    entity_class(
+                        f"{config_entry.title} {selector.name}",
+                        device,
+                        config_entry,
+                        f"{selector.short_name}_{config_entry.unique_id}",
+                        selector,
+                        coordinator,
+                    )
+                )
+
+    async_add_entities(entities)
+
+
+class XiaomiSelector(XiaomiCoordinatedMiioEntity, SelectEntity):
+    """Representation of a generic Xiaomi attribute selector."""
+
+    def __init__(self, name, device, entry, unique_id, selector, coordinator):
+        """Initialize the generic Xiaomi attribute selector."""
+        super().__init__(name, device, entry, unique_id, coordinator)
+        self._attr_icon = selector.icon
+        self._controller = selector
+        self._current_option = None
+
+    @property
+    def options(self):
+        """Return available options."""
+        return self._controller.options
+
+    @property
+    def current_option(self):
+        """Return the current option."""
+        return getattr(self, SERVICE_TO_METHOD[self._controller.service]["property"])
```

Comment:
Simplify this for now and remove all of the abstraction.

Reply:
Done